### PR TITLE
ISPN-15466 Passivation does not need to remove from store on load or …

### PR DESCRIPTION
**Pull request by wburns**
…write

https://issues.redhat.com/browse/ISPN-15466

### CODE REVIEW

File: core/src/test/java/org/infinispan/persistence/PassivationFunctionalTest.java [hunk 1]
```diff
@@ -154,30 +165,30 @@ public void testRemoveAndReplace() throws PersistenceException {
       assertNotInCacheAndStore("k1");
 
       assert cache.put("k2", "v2-NEW").equals("v2");
-      assertInCacheNotInStore("k2", "v2-NEW");
+      assertInCacheAndInStore("k2", "v2-NEW");
```
Comment (jabolina) [inline core/src/test/java/org/infinispan/persistence/PassivationFunctionalTest.java:168]: I am a bit confused. Shouldn't the update occurr only on cache and not in store?

**Comment by jabolina** [comment]
We may have it already, but should we add a test with a restart to ensure the store's values are updated?

File: core/src/test/java/org/infinispan/persistence/PassivationFunctionalTest.java [hunk 2]
```diff
@@ -64,6 +65,16 @@ public void afterMethod() throws PersistenceException {
...
    }
 
+   private void assertInCacheAndInStore(Object key, Object value) throws PersistenceException {
+      assertInCacheAndInStore(key, value, -1);
+   }
+
+   private void assertInCacheAndInStore(Object key, Object value, long lifespanMillis) throws PersistenceException {
+      InternalCacheValue se = cache.getAdvancedCache().getDataContainer().get(key).toInternalCacheValue();
+      testStoredEntry(se, value, lifespanMillis, "Cache", key);
+      assertTrue(store.contains(key));
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/PassivationFunctionalTest.java:75]: We could, I would just have to add an additional argument and update all the references. I can do that though.

**Comment by wburns** [comment]
> We may have it already, but should we add a test with a restart to ensure the store's values are updated?

There should be some already, but I updated the PassivatePersistentTest to have it explicitly now.

**Comment by wburns** [comment]
Also checking the StringStoreWithPooledConnectionFunctionalTest failure

**Review by jabolina — APPROVED** [review]
Looking good. I think we'll need some update on the docs.

**Comment by wburns** [comment]
> Looking good. I think we'll need some update on the docs.

Yeah, I was in the process of adding the doc changes. I actually cleaned up the passivation section in general as parts of it were outdated sadly.

**Comment by wburns** [comment]
Rebased to latest

**Pull request merged**